crates/node/src/networking/mod.rs

```diff
@@ -1,3 +1,54 @@
 pub mod download_manager;
 pub mod p2p;
+
+use eyre::{eyre, Result};
+use futures_util::TryStreamExt;
+use std::sync::Arc;
+use tokio::io::AsyncBufReadExt;
+use tokio_util::io::StreamReader;
+
 pub use p2p::P2P;
+
+use crate::storage::Database;
+
+pub struct WhitelistSyncer {
+    url: String,
+    database: Arc<Database>,
+}
+
+impl WhitelistSyncer {
+    pub fn new(url: String, database: Arc<Database>) -> Self {
+        Self { url, database }
+    }
+
+    pub async fn sync(&self) -> Result<()> {
+        let url = reqwest::Url::parse(&self.url)?;
+        let client = reqwest::ClientBuilder::new().gzip(true).build()?;
```

Comment:
The gzip here is specifically put on the transport layer. Originally I planned to have the file itself compressed, but then realized that it doesn't really add any benefits, but makes tracking of individual changes more difficult. For the use case here, it's enough that the transport itself is compressed.

Regarding the signing, I'm not sure what kind of added security it provides in devnet situation?

Reply:
Even for devnet people can want to do some bad thing. I didn't know the source and though it was a http connection. 
I see that the list came from github. So if we put the url in the code we avoid a configuration issue, and we are sure of the source. It doesn't seem that the source will change soon.

Reply:
The devnet people can do "anything" on their own node. They can use `gevulot peer <key> whitelist` or they can directly modify their DB, but that's not very useful because other nodes won't get the same changes and therefore it's useless effectively. If the file on our website gets compromised, it pretty much means that our GitHub access control has been compromised and at that point, the signature in the file doesn't really mean anything when the gevulot node implementation can be changed from the upstream as well.

Would there be some other aspects that could still benefit from the signature?